zcash_client_sqlite/src/wallet/init/migrations/add_utxo_account.rs

```diff
@@ -123,3 +128,87 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
         panic!("Cannot revert this migration.");
     }
 }
+
+#[cfg(feature = "transparent-inputs")]
+fn get_transparent_receivers<P: consensus::Parameters>(
+    conn: &rusqlite::Connection,
+    params: &P,
+    account: AccountId,
+) -> Result<HashMap<TransparentAddress, NonHardenedChildIndex>, SqliteClientError> {
+    let mut ret = HashMap::new();
+
+    // Get all UAs derived
+    let mut ua_query = conn
+        .prepare("SELECT address, diversifier_index_be FROM addresses WHERE account = :account")?;
+    let mut rows = ua_query.query(named_params![":account": u32::from(account)])?;
+
+    while let Some(row) = rows.next()? {
+        let ua_str: String = row.get(0)?;
+        let di_vec: Vec<u8> = row.get(1)?;
+        let mut di: [u8; 11] = di_vec.try_into().map_err(|_| {
+            SqliteClientError::CorruptedData(
+                "Diverisifier index is not an 11-byte value".to_owned(),
+            )
+        })?;
+        di.reverse(); // BE -> LE conversion
+
+        let ua = Address::decode(params, &ua_str)
+            .ok_or_else(|| {
+                SqliteClientError::CorruptedData("Not a valid Zcash recipient address".to_owned())
+            })
+            .and_then(|addr| match addr {
+                Address::Unified(ua) => Ok(ua),
+                _ => Err(SqliteClientError::CorruptedData(format!(
+                    "Addresses table contains {} which is not a unified address",
+                    ua_str,
+                ))),
+            })?;
+
+        if let Some(taddr) = ua.transparent() {
+            let di_short = DiversifierIndex::from(di).try_into();
+            if let Ok(di_short) = di_short {
+                if let Some(index) = NonHardenedChildIndex::from_index(di_short) {
+                    ret.insert(*taddr, index);
+                }
+            }
+        }
```

Comment:
As I mentioned above, this function isn't expected to have to keep pace with the public function. This is a snapshot in time of how the function used to behave. It will surely get out of date w.r.t. the public function, but so long as it does the migration correctly, it should be adequate. Did you notice in the changes made in main that something would break migrations?

Reply:
In this particular case, the rationale here would be that the fix you made in the earlier PR was a fix to this migration, and should be kept.

Reply:
fresh copy applied